apps/web/components/eventtype/EventAvailabilityTab.tsx

```diff
@@ -281,25 +285,177 @@ const EventTypeSchedule = ({ eventType }: { eventType: EventTypeSetup }) => {
   );
 };
 
-const UseCommonScheduleSettingsToggle = ({ eventType }: { eventType: EventTypeSetup }) => {
+const TeamMemberSchedule = ({
+  host,
+  index,
+  teamMembers,
+}: {
+  host: Host;
+  index: number;
+  teamMembers: TeamMembers;
+}) => {
+  const { t } = useLocale();
+
+  const formMethods = useFormContext<FormValues>();
+  const { setValue, getValues, getFieldState } = formMethods;
+
+  const { data, isPending } = trpc.viewer.availability.schedule.getAllSchedulesByUserId.useQuery({
+    userId: host.userId,
+  });
+
+  const schedules = data?.schedules;
+  const options = schedules?.map((schedule) => ({
+    value: schedule.id,
+    label: schedule.name,
+    isDefault: schedule.isDefault,
+    isManaged: false,
+  }));
+
+  //Set to defaultSchedule if Host Schedule is not previously selected
+  const scheduleId = getValues(`hosts.${index}.scheduleId`);
+  const value = options?.find((option) =>
+    scheduleId
+      ? option.value === scheduleId
+      : option.value === schedules?.find((schedule) => schedule.isDefault)?.id
+  );
+
+  if (!scheduleId && !getFieldState(`hosts.${index}.scheduleId`).isDirty) {
+    setValue(`hosts.${index}.scheduleId`, value?.value || null, { shouldDirty: true });
+  }
```

Comment:
This is still sending null to backend if you add host and save directly without making changes in the availability tab? Not sure what this is achieving? Imo this is not required. In case no scheduleId is being selected, aren't we defaulting to users default schedule anyway?


https://github.com/user-attachments/assets/99174cc4-bce3-4dc5-a1fe-7d82d862bb09

Reply:
This may occur if host is added in Assignment Tab and form is saved without visiting Availability Tab.
Alternatively, this can be moved to submit handler, but there we will not have AvailabilityOptions to set the default schedule id. As  now we are confined AvailabilityOptions to TeamMemberSchedule component only and removed in Host. And we cant rely on .team.member.user.defaultScheduleId as this not always populated. (had checked this in trpc endpoint).

Anyhow the functionality of getUserAvailability() will still pick the default Schedule, as this how its updated at backend. So the event booking will not break.

Will update to handle this scenario, but not sure if this line can be completely removed. Will check and update.